### PR TITLE
Implemented support for partials.

### LINKED ISSUE #1: Support for partials
It doesn't look like there is currently support for partials `{{> next_more}}`, or am I missing something?

### PR DESCRIPTION
**Pull request by h3rald**
This should fix #1.

### CODE REVIEW

**Comment by watzon** [comment]
Anything holding this up @fenekku ?

**Comment by fenekku** [comment]
> Anything holding this up @fenekku ?

Vacation mostly :), but also I think I never quite understood the "disk partials" use and use case which was why I didn't implement it if I recall. I implemented the "in-memory" partials however which made sense to me. 

I did it by not having an additional `partials` parameter because I didn't quite see the need for it - the API to make work in other implementations have always seemed clunky to me. Sure it would help to differentiate between a filename and a string but the computer can probably do it for you. The questions "is that a filename's path?" and "What is the relative location of that path?" just added more complexity. I wanted something simpler, but again it is probably because I never understood the full picture.  

Could you provide an example of file structure, templates, contexts and calls to illustrate the usage of disk partials? Once it makes sense I would be happy to merge!

**Comment by h3rald** [comment]
Hi! 

Sure. I am currently building a small, self-contained static site generator ([hastysite](https://github.com/h3rald/hastysite)) in nim, and I am currently using mustache as templating engine.

I implemented partials support because I needed a way to reuse things like header, footer etc. in (potentially) several page-level templates. [Here](https://github.com/h3rald/min/tree/master/site/templates) you can find some example templates -- in particular the page.mustache template references the others. This is a pointless example because I only have one top-level template, but I could create others and reuse the same partials.

The actual render call within hastysite is [here](https://github.com/h3rald/hastysite/blob/master/hastysite.nim#L311). In this case, I am assuming that all templates will be within the same folder (let's say it's "good enough" for me).

**Comment by fenekku** [comment]
Cool! 

Let me look into this/try this out because I also used partials for my personal static site generator (but clearly I didn't have 'disk partials'). I had the generator program read specific templates from disk and assign them to context variables which were then injected as partials. I had complete control this way.

File: src/moustachu.nim
Comment (fenekku) [inline src/moustachu.nim]: Given that `os` is imported at the top of the file, we shouldn't have to import it again.

File: src/moustachu.nim
Comment (fenekku) [inline src/moustachu.nim]: Let's change `pwd` to `partials_dir` since the directory containing the partials files is not necessarily the present working directory.

File: src/moustachu.nim
Comment (fenekku) [inline src/moustachu.nim]: I still like to be able to pass in-memory moustache strings in my context variables. And those should take precedence over disk partials.

We can have something like:

```nim
var partialTemplate = contextStack.lookupString(token.value)
if partialTemplate == "":
  partialTemplate = readFile(partials_dir.joinPath(token.value & ".mustache"))
```

File: src/moustachu.nim
Comment (fenekku) [inline src/moustachu.nim]: `pwd` -> `partials_dir`

File: src/moustachu.nim
Comment (fenekku) [inline src/moustachu.nim]: personal preference: `parentDir(tmplateFilename)` (or without the ending `()`)

**Review by fenekku — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Once the comments are addressed I will merge this in.

**Comment by fenekku** [comment]
Also don't forget to bump the version!

**Comment by h3rald** [comment]
@fenekku Sorry for the delay -- I think I implemented the changes you requested, please take a look :)
I bumped the version to 0.11.0... not sure if you follow semantic versioning, but I thought that this could be considered a new feature rather than a bugfix after all, but again up to you!

**Review by fenekku — APPROVED** [review]
Thank you for your contribution @h3rald !

**Pull request merged**
